File: src/main/java/com/shvants/UrlShorter/entities/Link.java
package com.shvants.UrlShorter.entities;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

public class Link {

    private Long linkId;
    private String url;
    private String shortUrl;
    private String description;
    private AtomicInteger clickAmount;
    private List<Tag> tagList;

    public Link(String url) {
        this.url = url;
    }

    public Link(String url, String description) {
        this(url);
        this.description = description;
    }

    public Link(String url, Tag ... tags) {
        this(url);
        this.tagList = new ArrayList<>(Arrays.asList(tags));
    }

    public Link(String url, String description, Tag ... tags) {
        this(url, description);
        this.tagList = new ArrayList<>(Arrays.asList(tags));
    }
}

File: src/urlshortener-ui/src/entity/link.ts
export class Link{

  linkId: number;
  url: string;
  shortUrl: string;
  tags: any;
  description: string;


  // constructor() {
  // }

// constructor(url: string) {
  //   this.url = url;
  // }

constructor(/*linkId: number,*/ url: string, shortUrl: string, description: string) {
    // this.linkId = linkId;
    this.url = url;
    this.shortUrl = shortUrl;
    this.description = description;
  }


}
